src/conversions.rs

```diff
@@ -481,12 +501,15 @@ impl<T: FromJSValConvertible> FromJSValConvertible for Option<T> {
     unsafe fn from_jsval(cx: *mut JSContext,
                          value: HandleValue,
                          option: T::Config)
-                         -> Result<Option<T>, ()> {
+                         -> Result<ConversionResult<Option<T>>, ()> {
         if value.get().is_null_or_undefined() {
-            Ok(None)
+            Ok(ConversionResult::Failure(Cow::Owned("Value is null or undefined.".to_owned())))
```

Comment:
@GuillaumeGomez: I'm pretty sure being able to return a borrowed cow is the reason for @nox to propose it, can you change it in the relevant places?